src/cmd/rm.go

```diff
@@ -70,6 +70,8 @@ func rm(cmd *cobra.Command, args []string) error {
 		return nil
 	}
 
+	var encounteredError bool = false
```

Comment:
Yes, I had noticed this regression while reviewing the Go rewrite in https://github.com/containers/toolbox/pull/318 I chose to ignore it because I couldn't think of a nice way to do it back then and didn't realize that it would have a negative effect on the tests.

It is a bit ugly to have each command directly call `os.Exit` this way. It's generally nicer to have a single return or exit point. In this case, we'd somehow propagate the error (and/or the exit code) up the function call stack, and have `Execute` in `cmd/root.go` handle it accordingly. However, I don't know off-hand how to achieve that with Cobra, but it should be possible.

In the worst case, given that, as far as we know, nobody is using `/usr/bin/toolbox` as an API from another program, I'd be happier to tweak the test suite to not expect a non-zero exit code, as a short-term temporary measure. Doesn't the test suite already look for the expected error message strings? If so, then I can live with it, while we figure out something nicer.

Reply:
Fair enough. I'll drop those two commits and adjust the test cases.